websites/R/Richup.io/presence.ts

```diff
@@ -0,0 +1,98 @@
+const presence = new Presence({
+		clientId: "1067910906912194590",
+	}),
+	pages: { [k: string]: string } = {
+		"/settings": "Settings",
+		"/store/player-appearance": "Player Appearance",
+		"/store/board-maps": "Board Maps",
+		"/store/upgrades": "Upgrades",
+		"/store/avatars": "Profile Pictures",
+		"/store/coins": "Coins",
+	};
+
+let startTimestamp: number;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "rio-logo",
+		},
+		{ pathname, href } = location,
+		[showJoinRoomButton, showProfileButton] = await Promise.all([
+			presence.getSetting("join-room-button"),
+			presence.getSetting("profile-button"),
+		]);
+
+	if (pathname.includes("/store")) {
+		presenceData.details = "Store";
+		if (pages[pathname]) presenceData.state = pages[pathname];
+	} else if (pathname.includes("/room")) {
+		let latestInfo = document.querySelector(
+				"#app > div:nth-child(4) > div > div:nth-child(2) > div > div > :first-child > :first-child > :last-child > :first-child"
+			)?.children?.[0]?.textContent,
+			players = [];
+
+		presenceData.details = "🎮 In a room";
+
+		if (latestInfo) {
+			const playerList = document.querySelectorAll(
+				"#app > div:nth-child(4) > div > div:nth-child(3) > div > :first-child > div > div > :not(:first-child) > div:nth-child(2) > div:first-child"
+			);
+
+			for (const element of playerList) players.push(element.textContent);
+
+			players = [...players]
+				.sort((a, b) => a.length - b.length)
+				.filter(Boolean);
+
+			/* Replace room ID */
+			latestInfo = latestInfo.replace(pathname.split("/")[2], "");
+
+			/* Replace player names */
+			for (const player of players)
+				latestInfo = latestInfo.replace(new RegExp(player, "g"), "A player");
+
+			presenceData.state = latestInfo ? `🎲 ${latestInfo}` : "";
+		}
+
+		if (!startTimestamp) startTimestamp = Date.now();
+		presenceData.startTimestamp = startTimestamp;
+
+		const gameSettings = document
+			.querySelector("#app > div > div > div:nth-child(3)")
+			?.textContent?.includes("Game settings");
+
+		if (showJoinRoomButton === true && gameSettings === true) {
```

Comment:
Unnecessary comparison of boolean variables.
```suggestion
		if (showJoinRoomButton && gameSettings) {
```